src/nfa/nfa-inclusion.cc

```diff
@@ -152,25 +188,44 @@ bool Mata::Nfa::Algorithms::is_incl_antichains(
 
 				if (is_subsumed) { continue; }
 
-				// prune data structures and insert succ inside
-				for (std::list<ProdStateType>* ds : {&processed, &worklist}) {
-					auto it = ds->begin();
-					while (it != ds->end()) {
-						if (subsumes(succ, *it)) {
-							auto to_remove = it;
-							++it;
-							ds->erase(to_remove);
-						} else {
-							++it;
-						}
-					}
-
-					// TODO: set pushing strategy
-					ds->push_back(succ);
-				}
+                // prune data structures and insert succ inside
+                //TODO: this code segfaults with deques instead of lists everywhere :(
+                // (Works on apple, not on Ubuntu. Removal invalidates iterators.)
+                // A fix could be removal by moving the last element there and decreasing the size
+                //for (std::list<ProdStateType>* ds : {&processed, &worklist}) {
+                //    auto it = ds->begin();
+                //    while (it != ds->end()) {
+                //        if (subsumes(succ, *it)) {
+                //            auto to_remove = it;
+                //            ++it;
+                //            ds->erase(to_remove);
+                //        } else {
+                //            ++it;
+                //        }
+                //    }
+
+                //    // TODO: set pushing strategy
+                //    ds->push_back(succ);
+                //}
+
+                for (std::deque<ProdStateType>* ds : {&processed, &worklist}) {
+                    for (size_t it = 0; it < ds->size(); ++it) {
```

Comment:
As usual: extract the `size()` if it is possible.

Reply:
not my foult, ask Ondra :)

Reply:
Aha but I added my favourite trick so that size changes during the iteration :).